Add git_pillar integration tests

terminalmage opened the pull request:
This PR does the following:

1. Removes the git_pillar configuration from the integration suite's master config file.
2. Removes the tests which reference the pillar data provided by the git ext_pillar configuration in the master config file. These are no longer necessary because a) the data they use has been removed, b) the functionality being tested is provided in the new integration tests added by this PR, or c) it directly tests the legacy git_pillar, which is slated for removal in Oxygen and has not been actively developed in a couple years.
3. Adds a new directory in the integration suite called ``pillar`` for ext_pillar tests, and also updates runtests.py so that ext_pillar tests can be invoked as a group in the same way that runner, module, etc. tests can.
4. Adds integration tests for git_pillar. Both GitPython and pygit2 are tested, each with SSH, HTTP, and HTTP w/basic auth.

terminalmage committed:
Remove legacy git_pillar tests

Also remove git_pillar configuration from the test suite's master
configuration file.

terminalmage committed:
Add git_pillar integration tests

This also adds a new section to the runtests.py so that ext_pillar
integration tests can be run separately.

salt-jenkins commented:
@terminalmage, thanks for your PR! By analyzing the history of the files in this pull request, we identified @rallytime, @pass-by-value and @Vye to be potential reviewers.

terminalmage commented:
I've set the ``Pending Changes`` label to note that this should not be merged before we get https://github.com/saltstack/salt-jenkins/pull/299 merged.

terminalmage committed:
Prepare git_pillar test code for adding HTTP tests

This further abstracts some of the setup and teardown code so it can be
used for git-over-http tests.

It also moves the code that was originally added to the archive
state integration tests to create a local http server into
salt.support.helpers so that it can be more easily and portably used.

terminalmage commented:
Go Go Jenkins!

terminalmage commented:
SSH tests are failing for pygit2, but this is likely due to the fact that EPEL7 currently has a version mismatch for pygit2 and libgit2. The libgit2 package was updated to the 0.24 release cycle while pygit2 remained at 0.21.4.

There is a build with enough karma to be pushed to stable, but the maintainer has yet to push it:

https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2017-d00d781835

I will give it another two days (at which time the package will be eligible for push irrespective of karma) and contact the maintainer to try to get it pushed to stable.

terminalmage committed:
Add GitPython HTTP git_pillar tests

terminalmage committed:
Add HTTP git_pillar integration tests for pygit2

Also work around ssh auth issues in Ubuntu

terminalmage committed:
Skip Pygit2 tests until EPEL fixes pygit2/libgit2 version mismatch

We can revert this commit once they get the updated pygit2 pushed to
EPEL stable.

terminalmage commented:
OK, I took a look at the pygit2 errors, and at the salt-runtests.log, and confirmed that the failures are due to the current version mismatch in EPEL. They fail in the same way, with the same errors in the salt-runtests.log, when I downgrade pygit2 to the version available in the stable EPEL repo.

In the meantime [this build](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2017-d00d781835) is pushed to stable, I have temporarily skipped the pygit2 tests on EL7 with a commit I have just pushed to this PR.

terminalmage commented:
OK, tests are all passing now. This should be good to merge.
